DatadogSessionReplay/Tests/Mocks/ReflectionMocks.swift

```diff
@@ -0,0 +1,114 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+#if os(iOS)
+@testable import DatadogSessionReplay
+import TestUtilities
+import SwiftUI
+import CoreGraphics
+
+private let bytesPerPixel = 4
+
+private extension Int {
+    static func mockImageDimension() -> Int {
+        let min = 1, max = 10_000
+        return Int.random(in: min...max)
+    }
+}
+
+/// A mock implementation of `CGImage` for unit tests.
+/// - Note: This mock creates a dummy RGBA image of configurable size.
+/// - Warning: Ensure the width is within acceptable limits to avoid memory issues.
+struct MockCGImage: RandomMockable {
+    let cgImage: CGImage
+
+    init(cgImage: CGImage) {
+        self.cgImage = cgImage
+    }
+
+    static func mockRandom() -> MockCGImage {
+        return MockCGImage(
+            cgImage: mockWith()
+        )
+    }
+
+    static func mockWith(
+        width: Int = .mockImageDimension()
+    ) -> CGImage {
+        precondition(width > 0, "Width must be greater than 0")
+
+        let totalBytes = width * width * bytesPerPixel
+
+        // Allocate memory for pixel data
+        guard let data = malloc(totalBytes) else {
+            fatalError("Failed to allocate memory for pixel data")
+        }
+
+        // Free the allocated memory when the function exits
+        defer { free(data) }
+
+        // Fill the allocated memory with dummy data
+        memset(data, 255, totalBytes)
```

Comment:
/suggestion They might be less low level way of generating bitmap, like drawing a fill color! This could enable having true random image by randomising the color. WDYT?

Reply:
Yes, that's a good idea!